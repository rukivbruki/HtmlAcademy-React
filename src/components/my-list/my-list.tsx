// Libs
import * as React from 'react';
import {connect} from "react-redux";
// Utils
import {FilmsType} from '../../types.js';
import {Operation as FavoritesOpearion} from "../../reducer/favorites/favorites.js";
import {getFavorites} from "../../reducer/favorites/selectors.js";
// Components
import Logo from "../logo/logo";
import Footer from "../footer/footer";
import Avatar from '../avatar/avatar';
import MovieCard from "../movie-card/movie-card";

interface Props {
  favorites?: FilmsType;
  loadFavorites: () => void;
}

class MyList extends React.PureComponent<Props> {
  constructor(props: Props) {
    super(props);
  }

  componentDidMount() {
    const {loadFavorites} = this.props;
    loadFavorites();
  }

  render() {
    const {favorites = []} = this.props;

    return (
      <div className="user-page">
        <header className="page-header user-page__head">
          <Logo />

          <h1 className="page-title user-page__title">My list</h1>

          <div className="user-block">
            <Avatar />
          </div>
        </header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <div className="catalog__movies-list">
            {favorites.map(({id, name, genre, previewImage, videoLink}) => {
              return <MovieCard key={id} id={id} name={name} genre={genre} previewImage={previewImage} videoLink={videoLink}/>;
            })}
          </div>
        </section>

        <Footer />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  favorites: getFavorites(state)
});

const mapDispatchToProps = (dispatch) => ({
  loadFavorites() {
    dispatch(FavoritesOpearion.loadFavorites());
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
