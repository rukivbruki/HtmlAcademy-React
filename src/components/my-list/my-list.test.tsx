import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {MyList} from "./my-list";

const mockStore = configureStore([]);

const props = {
  loadFavorites: () => void 0
};

const favorites = [
  {
    id: 1,
    name: `test`
  },
  {
    id: 2,
    name: `test`
  },
  {
    id: 3,
    name: `test`
  },
];

it(`MyList renders correctly`, () => {
  const store = mockStore({
    [NameSpace.FAVORITES]: {
      favorites,
    },
    [NameSpace.USER]: {
      user: {
        id: 1,
        email: `test@email.com`,
        name: `test user`,
        avatarUrl: `avatar`
      }
    },
  });

  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <MyList {...props} />
        </BrowserRouter>
      </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
