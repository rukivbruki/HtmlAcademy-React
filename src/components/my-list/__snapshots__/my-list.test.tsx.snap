// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="sign-in__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    />
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
